### ISSUE TITLE
Pipewire 0.3.82 removes the Pipewire server entirely and gives a dummy exit. The new version 0.3.84 creates the exact same problem 

### ISSUE THREAD

**Issue by diabolicbg** [pipewire-debian/pipewire-debian#99]
**If you Completed all troubleshoot options, check below**     
- [* ]  Did you Follow all installation instructions carefully from [here](../wiki/#1-ppa-configuration)?
- [* ]  Did you check All troubleshoot options from [here](../wiki/Troubleshooting)?

**Desktop (please complete the following information):**
 - Distro: Linux Mint
 - Desktop: MATE
 - Version: 21.2
 - PipeWire Related packages Version: PipeWire 0.3.79
 - Kernel version: 5.15.0-87

**Describe the bug**
Pipewire 0.3.82 removes the Pipewire server entirely and gives a dummy exit immediately after system restart. I couldn't find a way to activate it and had to revert to the old snapshot with Timeshift.
```
Audio:
  Device-1: NVIDIA GP107GL High Definition Audio vendor: Gigabyte driver: snd_hda_intel v: kernel
    pcie: speed: 8 GT/s lanes: 16 bus-ID: 01:00.1 chip-ID: 10de:0fb9
  Device-2: Creative Labs Sound Core3D [Sound Blaster Recon3D / Z-Series] driver: snd_hda_intel
    v: kernel pcie: speed: 2.5 GT/s lanes: 1 bus-ID: 04:00.0 chip-ID: 1102:0012
  Sound Server-1: ALSA v: k5.15.0-87-generic running: yes
  Sound Server-2: PulseAudio v: 15.99.1 running: no
```
I should clarify that PulseAudio was pre-disabled right after installing the system because it didn't work properly with my Creative Labs Sound Blaster and only played stereo with no surround sound selection.
```
pactl info
Server String: /run/user/1000/pulse/native
Library Protocol Version: 35
Server Protocol Version: 35
Is Local: yes
Client Index: 15
Tile Size: 65472
User Name: ivo
Host Name: ivo-home-pc
Server Name: pulseaudio
Server Version: 15.99.1
Default Sample Specification: s16le 2ch 44100Hz
Default Channel Map: front-left,front-right
Default Sink: auto_null
Default Source: alsa_input.pci-0000_04_00.0.analog-stereo
```
This is system information from when it works. It can be seen that Sound Server-2: PulseAudio v: 15.99.1 running: no.
```
Audio:
  Device-1: NVIDIA GP107GL High Definition Audio vendor: Gigabyte driver: snd_hda_intel v: kernel
    pcie: speed: 8 GT/s lanes: 16 bus-ID: 01:00.1 chip-ID: 10de:0fb9
  Device-2: Creative Labs Sound Core3D [Sound Blaster Recon3D / Z-Series] driver: snd_hda_intel
    v: kernel pcie: speed: 2.5 GT/s lanes: 1 bus-ID: 04:00.0 chip-ID: 1102:0012
  Sound Server-1: ALSA v: k5.15.0-87-generic running: yes
  Sound Server-2: PulseAudio v: 15.99.1 running: no
  Sound Server-3: PipeWire v: 0.3.79 running: yes
```
```
pactl info
Server String: /run/user/1000/pulse/native
Library Protocol Version: 35
Server Protocol Version: 35
Is Local: yes
Client Index: 153
Tile Size: 65472
User Name: ivo
Host Name: ivo-home-pc
Server Name: PulseAudio (on PipeWire 0.3.79)
Server Version: 15.0.0
Default Sample Specification: float32le 2ch 48000Hz
Default Channel Map: front-left,front-right
Default Sink: alsa_output.pci-0000_04_00.0.analog-surround-41
Default Source: alsa_input.pci-0000_04_00.0.analog-stereo
```
In my first quote I gave the same info where Sound Server-3: PipeWire is absent. That's why I wrote that the upgrade actually removes it. Before installation it uninstalls 5-6 packages that are probably important for the server to work.                            
.                                          
.                                          
.

**Comment by diabolicbg** [comment]
Hey souravdas142 what are you doing? Do you even read the trouble alerts?
The new version 0.3.84 creates the exact same problem - it removes all files associated with PipeWire and does not restore them after installation. 
Again I ended up with no system sound and a dummy output. 
Removed files: 
![2023-11-04_09-54](https://github.com/pipewire-debian/pipewire-debian/assets/149315793/807c0a4d-db4a-4ee0-b8ce-3ae22aefbebf)

**Comment by souravdas142** [comment]
Sorry Can't reproduce here...

> Before installation it uninstalls 5-6 packages that are probably important for the server to work.

Can you list down those packages ? 

These are half information, I could not do anything. Mint based on ubuntu 22.04, here for ubuntu version, there is no issue.
Please provide more information.

**Comment by diabolicbg** [comment]
All the packages are listed in the picture in my last post.

**Comment by souravdas142** [comment]
Hmmm, could you please run `sudo apt-get update && sudo apt-get upgrade` from the terminal ? Don't give actual upgrade. I need  the output only what it describes ?

**Comment by diabolicbg** [comment]
```
sudo apt-get update && sudo apt-get upgrade
[sudo] password for ivo:          
Поп:1 http://archive.ubuntu.com/ubuntu jammy InRelease
Поп:2 http://archive.ubuntu.com/ubuntu jammy-updates InRelease                 
Поп:3 http://archive.ubuntu.com/ubuntu jammy-backports InRelease               
Поп:4 https://dl.winehq.org/wine-builds/ubuntu jammy InRelease                 
Поп:5 https://ppa.launchpadcontent.net/cdemu/ppa/ubuntu jammy InRelease        
Поп:6 http://security.ubuntu.com/ubuntu jammy-security InRelease               
Поп:7 https://ppa.launchpadcontent.net/jonmagon/crow-translate/ubuntu jammy InRelease
Поп:8 https://ppa.launchpadcontent.net/pipewire-debian/pipewire-upstream/ubuntu jammy InRelease
Игн:9 https://mirrors.netix.net/LinuxMint/linuxmint-repo victoria InRelease
Поп:10 https://mirrors.netix.net/LinuxMint/linuxmint-repo victoria Release
Четене на списъците с пакети... Готово 
Четене на списъците с пакети... Готово
Изграждане на дървото със зависимости... Готово
Четене на информацията за състоянието... Готово      
Изчисляване на актуализацията... Готово
The following packages will not be changed:
  gstreamer1.0-pipewire libnvidia-cfg1-530 libnvidia-cfg1-535
  libnvidia-common-535 libnvidia-compute-535 libnvidia-compute-535:i386
  libnvidia-decode-535 libnvidia-decode-535:i386 libnvidia-encode-535
  libnvidia-encode-535:i386 libnvidia-extra-535 libnvidia-fbc1-535
  libnvidia-fbc1-535:i386 libnvidia-gl-535 libnvidia-gl-535:i386 libopus0
  libpipewire-0.3-0 libpipewire-0.3-dev libpipewire-0.3-modules
  libspa-0.2-bluetooth libspa-0.2-dev libspa-0.2-jack libspa-0.2-modules
  nvidia-compute-utils-535 nvidia-dkms-535 nvidia-driver-530 nvidia-driver-535
  nvidia-kernel-common-535 nvidia-kernel-source-535 nvidia-utils-535 pipewire
  pipewire-alsa pipewire-bin pipewire-jack pipewire-pulse pipewire-tests
  xserver-xorg-video-nvidia-535
The following packages will be updated:
  pipewire-audio-client-libraries pipewire-locales

```
When I run the update from the update manager, I first get the remove packages message that your seeing in the screenshot. Then the installation itself takes 2-3 seconds. Finally after a system reboot I find that I don't have PipeWire.

**Comment by souravdas142** [comment]
> The following packages will not be changed:

This is strange... 

And also there is no remove list... 

could you check you are not holding any packages

please give the output of `sudo apt-get upgrade --with-new-pkgs`

**Comment by diabolicbg** [comment]
```
sudo apt-get upgrade --with-new-pkgs
[sudo] password for ivo:          
Четене на списъците с пакети... Готово
Изграждане на дървото със зависимости... Готово
Четене на информацията за състоянието... Готово      
Изчисляване на актуализацията... Готово
The following packages will not be changed:
  gstreamer1.0-pipewire libnvidia-cfg1-530 libnvidia-cfg1-535
  libnvidia-common-535 libnvidia-compute-535 libnvidia-compute-535:i386
  libnvidia-decode-535 libnvidia-decode-535:i386 libnvidia-encode-535
  libnvidia-encode-535:i386 libnvidia-extra-535 libnvidia-fbc1-535
  libnvidia-fbc1-535:i386 libnvidia-gl-535 libnvidia-gl-535:i386 libopus0
  libpipewire-0.3-0 libpipewire-0.3-dev libpipewire-0.3-modules
  libspa-0.2-bluetooth libspa-0.2-dev libspa-0.2-jack libspa-0.2-modules
  nvidia-compute-utils-535 nvidia-dkms-535 nvidia-driver-530 nvidia-driver-535
  nvidia-kernel-common-535 nvidia-kernel-source-535 nvidia-utils-535 pipewire
  pipewire-alsa pipewire-bin pipewire-jack pipewire-pulse pipewire-tests
  xserver-xorg-video-nvidia-535
The following packages will be updated:
  pipewire-audio-client-libraries pipewire-locales


```

**Comment by souravdas142** [comment]
very strange. do `sudo apt-get install pipewire-bin=0.3.84-1~ubuntu22.04`  and give output ?

did you check you are not holding any package from upgrade ?

**Comment by diabolicbg** [comment]
```
sudo apt-get install pipewire-bin=0.3.84-1~ubuntu20.04
[sudo] password for ivo:          
Read package lists... Done
Building the dependency tree... Done
Reading status information... Done      
The pipewire-bin package is a virtual package provided by:
  pipewire-bin:i386 0.3.48-1ubuntu3 (= 0.3.48-1ubuntu3)
You must explicitly select one to install.

E: Version "0.3.84-1~ubuntu20.04" of "pipewire-bin" not found


```

**Comment by souravdas142** [comment]
my bad see above edited

**Comment by diabolicbg** [comment]
```
sudo apt-get install pipewire-bin=0.3.84-1~ubuntu22.04
[sudo] password for ivo:          
Read package lists... Done
Building the dependency tree... Done
Reading status information... Done      
Some packages could not be installed. This could mean,
that you requested an impossible situation or if you are using the unstable
distribution, that some necessary packages have not yet been created or
have been moved from Incoming.
The following information may help to find a way out of the situation:

The following packages have unsatisfied dependencies:
 pipewire-bin : depends on: libpipewire-0.3-modules (= 0.3.84-1~ubuntu22.04) but it will not be installed
                Depends on: pipewire-pulse
                Recommended: wireplumber but it won't be installed either
                                           pipewire-media-session but it will not be installed
E: Failure to correct problems, you have broken packages retained.

```
Strange... The package manager does not show any corrupted packages.

**Comment by souravdas142** [comment]
now do the same for libpipewire-0.3-modules also... its a chain of packages will not going to be installed. find the root package which has the the problem.

**Comment by diabolicbg** [comment]
I don't understand exactly what I have to do. You want me to execute the command:
`sudo apt-get install libpipewire-0.3-modules=0.3.84-1~ubuntu22.04`
Did I understand correctly?

**Comment by diabolicbg** [comment]
```
sudo apt-get install libpipewire-0.3-modules=0.3.84-1~ubuntu22.04
[sudo] password for ivo:          
Reading package lists... Done
Building the dependency tree... Done
Reading status information... Done      
Some packages could not be installed. This could mean,
that you requested an impossible situation or if you are using the unstable
distribution, that some necessary packages have not yet been created or
have been moved from Incoming.
The following information may help to find a way out of the situation:

The following packages have unsatisfied dependencies:
 libpipewire-0.3-modules : Depends on: libopus0 (>= 1.3.1-2~) but will be installed 1.3.1-0.1build2
E: Failure to correct problems, you have broken packages retained.

```

**Comment by diabolicbg** [comment]
```
sudo apt-get install pipewire-pulse=0.3.84-1~ubuntu22.04
Reading the package lists... Done
Building the dependency tree... Done
Reading the status information... Done      
Some packages could not be installed. This could mean,
that you requested an impossible situation or if you are using the unstable
distribution, that some necessary packages have not yet been created or
have been moved from Incoming.
The following information may help to find a way out of the situation:

The following packages have unsatisfied dependencies:
 pipewire : depends on: libpipewire-0.3-modules (= 0.3.84-1~ubuntu22.04) but it will not be installed
            Depends on: pipewire-bin (= 0.3.84-1~ubuntu22.04)
E: Failure to correct problems, you have broken packages on hold.


```

**Comment by diabolicbg** [comment]
```
udo apt-get install wireplumber=0.3.84-1~ubuntu22.04
Reading the package lists... Done
Building the dependency tree... Done
Reading the status information... Done      
The package wireplumber is not available, but is in the dependency list of another package.
This could mean that the package is missing, out of date, or available
only from another source

E: Version "0.3.84-1~ubuntu22.04" of "wireplumber" was not found

```

**Comment by diabolicbg** [comment]
I will now go into "Advanced Options" on loading and try to fix from there if there are broken packages. 

Several packages were removed, 6 updated and this led to the same result - no PipeWire server. 
I've put the system back with Timeshift and I'm left waiting for advice.

**Comment by souravdas142** [comment]
> The following packages have unsatisfied dependencies:
> libpipewire-0.3-modules : Depends on: libopus0 (>= 1.3.1-2~) but will be installed 1.3.1-0.1build2
>E: Failure to correct problems, you have broken packages retained.


`libopus` is the suspicious package. can you check what is the latest version of it ? and why is not installable ?

**Comment by diabolicbg** [comment]
I would, but I don't know how. Please write me the commands!

**Comment by diabolicbg** [comment]
No newer file libopus0 - https://packages.ubuntu.com/jammy/libopus0

**Comment by souravdas142** [comment]
what is the output of `apt show libopus0` ?

**Comment by diabolicbg** [comment]
```
apt show libopus0
Package: libopus0
Version: 1.4-1~ubuntu22.04
Priority: optional
Section: libs
Source: opus
Maintainer: Debian Multimedia Maintainers <debian-multimedia@lists.debian.org>
Installed-Size: 401 kB
Depends: libc6 (>= 2.29)
Suggests: opus-tools
Download-Size: 205 kB
APT-Sources: https://ppa.launchpadcontent.net/pipewire-debian/pipewire-upstream/ubuntu jammy/main amd64 Packages
Description: Opus codec runtime library
 The Opus codec is designed for interactive speech and audio transmission over
 the Internet. It is designed by the IETF Codec Working Group and incorporates
 technology from Skype's SILK codec and Xiph.Org's CELT codec.
 .
 It is intended to suit a wide range of interactive audio applications,
 including Voice over IP, videoconferencing, in-game chat, and even remote live
 music performances.  It can scale from low bit-rate narrowband speech to very
 high quality stereo music. The current features are:
 .
 Bit-rates from 6 kb/s 510 kb/s
 Sampling rates from 8 to 48 kHz
 Frame sizes from 2.5 ms to 60 ms
 Support for both constant bit-rate (CBR) and variable bit-rate (VBR)
 Audio bandwidth from narrowband to full-band
 Support for speech and music
 Support for mono and stereo
 Support for up to 255 channels (multistream frames)
 Dynamically adjustable bitrate, audio bandwidth, and frame size
 Good loss robustness and packet loss concealment (PLC)
 Floating point and fixed-point implementation
 .
 This package provides the Opus runtime library.

N: There is 1 additional record. Please use the '-a' switch to see it
```
```
apt show libopus0 -a
Package: libopus0
Version: 1.4-1~ubuntu22.04
Priority: optional
Section: libs
Source: opus
Maintainer: Debian Multimedia Maintainers <debian-multimedia@lists.debian.org>
Installed-Size: 401 kB
Depends: libc6 (>= 2.29)
Suggests: opus-tools
Download-Size: 205 kB
APT-Sources: https://ppa.launchpadcontent.net/pipewire-debian/pipewire-upstream/ubuntu jammy/main amd64 Packages
Description: Opus codec runtime library
 The Opus codec is designed for interactive speech and audio transmission over
 the Internet. It is designed by the IETF Codec Working Group and incorporates
 technology from Skype's SILK codec and Xiph.Org's CELT codec.
 .
 It is intended to suit a wide range of interactive audio applications,
 including Voice over IP, videoconferencing, in-game chat, and even remote live
 music performances.  It can scale from low bit-rate narrowband speech to very
 high quality stereo music. The current features are:
 .
 Bit-rates from 6 kb/s 510 kb/s
 Sampling rates from 8 to 48 kHz
 Frame sizes from 2.5 ms to 60 ms
 Support for both constant bit-rate (CBR) and variable bit-rate (VBR)
 Audio bandwidth from narrowband to full-band
 Support for speech and music
 Support for mono and stereo
 Support for up to 255 channels (multistream frames)
 Dynamically adjustable bitrate, audio bandwidth, and frame size
 Good loss robustness and packet loss concealment (PLC)
 Floating point and fixed-point implementation
 .
 This package provides the Opus runtime library.

Package: libopus0
Version: 1.3.1-0.1build2
Priority: optional
Section: libs
Source: opus
Origin: Ubuntu
Maintainer: Ubuntu Developers <ubuntu-devel-discuss@lists.ubuntu.com>
Original-Maintainer: Ron Lee <ron@debian.org>
Bugs: https://bugs.launchpad.net/ubuntu/+filebug
Installed-Size: 401 kB
Depends: libc6 (>= 2.29)
Suggests: opus-tools
Homepage: http://www.opus-codec.org
Task: ubuntu-desktop-minimal, ubuntu-desktop, ubuntu-desktop-raspi, kubuntu-desktop, xubuntu-core, xubuntu-desktop, lubuntu-desktop, ubuntustudio-desktop-core, ubuntustudio-desktop, ubuntukylin-desktop, ubuntu-mate-core, ubuntu-mate-desktop, ubuntu-budgie-desktop, ubuntu-budgie-desktop-raspi
Download-Size: 203 kB
APT-Manual-Installed: yes
APT-Sources: http://archive.ubuntu.com/ubuntu jammy/main amd64 Packages
Description: Opus codec runtime library
 The Opus codec is designed for interactive speech and audio transmission over
 the Internet. It is designed by the IETF Codec Working Group and incorporates
 technology from Skype's SILK codec and Xiph.Org's CELT codec.
 .
 It is intended to suit a wide range of interactive audio applications,
 including Voice over IP, videoconferencing, in-game chat, and even remote live
 music performances.  It can scale from low bit-rate narrowband speech to very
 high quality stereo music. The current features are:
 .
  Bit-rates from 6 kb/s 510 kb/s
  Sampling rates from 8 to 48 kHz
  Frame sizes from 2.5 ms to 60 ms
  Support for both constant bit-rate (CBR) and variable bit-rate (VBR)
  Audio bandwidth from narrowband to full-band
  Support for speech and music
  Support for mono and stereo
  Support for up to 255 channels (multistream frames)
  Dynamically adjustable bitrate, audio bandwidth, and frame size
  Good loss robustness and packet loss concealment (PLC)
  Floating point and fixed-point implementation
 .
 This package provides the Opus runtime library.
```

**Comment by souravdas142** [comment]
do `sudo apt-get install libopus0=1.4-1~ubuntu22.04` and give the output

**Comment by diabolicbg** [comment]
```
sudo apt-get install libopus=1.4-1~ubuntu22.04
[sudo] password for ivo:          
Reading package lists... Done
Building the dependency tree... Done
Reading status information... Done      
E: The libopus package cannot be found
```

**Comment by souravdas142** [comment]
do `sudo apt-get install libopus0=1.4-1~ubuntu22.04` and give the output 

previous was typo do this

**Comment by diabolicbg** [comment]
```
sudo apt-get install libopus0=1.4-1~ubuntu22.04
[sudo] password for ivo:          
Reading package lists... Done
Building the dependency tree... Done
Reading status information... Done      
The following packages were automatically installed and are no longer needed:
  fuseiso i965-va-driver:i386 intel-media-va-driver:i386
  libatk-bridge2.0-0:i386 libatk1.0-0:i386 libatspi2.0-0:i386 libcolord2:i386
  libcurl3-gnutls:i386 libepoxy0:i386 libgdbm-compat4:i386 libgdbm6:i386
  libglu1-mesa:i386 libgtk-3-0:i386 libieee1284-3:i386 libigdgmm12:i386
  libnspr4:i386 libnss3:i386 libodbc1:i386 libodbc1 libodbccr2 libodbccr2:i386
  libopengl0:i386 libopenjp2-7:i386 libpci3:i386 libpcsclite1:i386
  libperl5.34:i386 libpoppler-glib8:i386 libpoppler118:i386 librsvg2-2:i386
  librsvg2-common:i386 libsane1:i386 libsnmp40:i386 libva-drm2:i386
  libva-x11-2:i386 libva2:i386 libwrap0:i386 mesa-va-drivers:i386
  va-driver-all:i386 wine-staging-amd64
Use "sudo apt autoremove" to remove them.
Suggested packages:
  opus-tools
The following packages will be REMOVED:
  gstreamer1.0-plugins-base:i386 gstreamer1.0-plugins-good:i386
  libasound2-plugins:i386 libfaudio0:i386 libopus0:i386 libpulse0:i386
  libsdl2-2.0-0:i386 libsndfile1:i386 libwine:i386 wine-staging
  wine-staging-i386:i386 wine32:i386 winehq-staging winetricks
The following packages will be updated:
  libopus0
1 updated, 0 new installed, 14 to remove and 17 unchanged.
It is necessary to download 205 kB archives.
After this operation, 1146 MB of disk space will be freed.
```

**Comment by souravdas142** [comment]
now see there is no package related to `pipewire` . some problems may be on wine side packaging... This repo can't do anything. Better stick with stock version. This is **Recommend only for you.**

**Comment by diabolicbg** [comment]
OK, thanks!

**Comment by souravdas142** [comment]
Not fixable as other packages hard dependent on libopus some version. Hence closing.
